app/src/parser_special.c

```diff
@@ -72,13 +62,41 @@ parser_error_t parser_getItem_NativeTransfer(ExecutableDeployItem item, parser_c
 
     uint8_t new_displayIdx = displayIdx - item.UI_fixed_items;
 
-    if (new_displayIdx > item.UI_runtime_items || displayIdx < item.UI_fixed_items) {
-        return parser_unexpected_number_items;
-    }
     uint32_t dataLength = 0;
     uint8_t datatype = 255;
 
-    if(!app_mode_expert()){
+    if (new_displayIdx >= item.UI_runtime_items || displayIdx < item.UI_fixed_items) {
+        return parser_unexpected_number_items;
+    }
+
+    // generic
+    if (item.with_generic_args > 0){
+        if (new_displayIdx == 0) {
```

Comment:
Isn't it possible to squeeze in the entry point and name for the generic case? As I see those in the test vectors

Reply:
yeah, but it is not clear if for native transfer we would get the situation where we have to show a hash. i added this just for later cases where an unexpected arg different than, source, target, id and amount is found in the list of args..however there is not test vectors for this specific case. 
and actually I did not know that native transfer contain and contract call entry-point?